### PR TITLE
Upgrade dependencies and fix linting issues.

### PR DESCRIPTION
**Pull request by niemyjski**
**Update: This pr addresses dependencies, we have a plan to solve this in the next update.**

This should resolve all CJS import issues,

I think what was happening is that by default we were saying the main script was `index.js` (https://github.com/exceptionless/Exceptionless.JavaScript/blob/master/packages/browser/package.json#L27) and that all consumers should use this and tree shake. The problem with this is that we had a dependency on `stacktrace-js` and import for `stacktrace-js` so it would always be loaded as a cjs umd module. If you used the bundled version transpiled with esbuild it would have worked.

This pr has a prebuild action to convert the stacktrace-js dependency and it's dependencies to esm local file, the build then references this file instead of the cjs requires and then post build copies this transpiled file to the dist folder (tsc doesn't copy js files or any non ts file by default).

### CODE REVIEW

**Comment by ejsmith** [comment]
Interesting. Hope it works.

File: package.json [hunk 1]
```diff
@@ -16,28 +16,32 @@
   },
   "scripts": {
     "clean": "rimraf packages/*/dist example/*/dist",
-    "build": "lage build --no-cache --verbose --no-deps",
```
Comment (ejsmith) [inline package.json:19]: Nice, got rid of lage.

**Review by ejsmith — APPROVED** [review]
LGTM

Looks like you've gotten better at TypeScript since you originally did this, eh? :-)

**Comment by niemyjski** [comment]
I still need to come up with a solution for: https://github.com/stacktracejs/stacktrace.js/issues/230

**Review by niemyjski — COMMENTED** [review]
Reviewed all files.

**Pull request merged**